serde/src/lib.rs

```diff
@@ -45,6 +45,30 @@ impl fmt::Display for DeserializeError {
 
 impl Error for DeserializeError {}
 
+pub const I8_SIZE: usize = 1;
+pub const U8_SIZE: usize = 1;
+
+pub const I16_SIZE: usize = (16 + 6) / 7;
+pub const U16_SIZE: usize = (16 + 6) / 7;
+pub const I32_SIZE: usize = (32 + 6) / 7;
+pub const U32_SIZE: usize = (32 + 6) / 7;
+pub const I64_SIZE: usize = (64 + 6) / 7;
+pub const U64_SIZE: usize = (64 + 6) / 7;
```

Comment:
These are maximum sizes, I'd change it to explicitly say it in the name
```suggestion
pub const MAX_I8_SIZE: usize = 1;
pub const MAX_U8_SIZE: usize = 1;

pub const MAX_I16_SIZE: usize = (16 + 6) / 7;
pub const MAX_U16_SIZE: usize = (16 + 6) / 7;
pub const MAX_I32_SIZE: usize = (32 + 6) / 7;
pub const MAX_U32_SIZE: usize = (32 + 6) / 7;
pub const MAX_I64_SIZE: usize = (64 + 6) / 7;
pub const MAX_U64_SIZE: usize = (64 + 6) / 7;
```